src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpResponseTask.cs

```diff
@@ -80,7 +80,7 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
 
             foreach (var header in headers)
             {
-                response.Headers.Add(header);
+                response.Headers.Append(header.Key, header.Value);
```

Comment:
Why it's safer?

Reply:
I saw it while testing dotnet 8 preview where new analyzers warn on it, and then I was not able to compile. In fact the warning was in `ModularTenantContainerMiddleware` on.

    httpContext.Response.Headers.Add(HeaderNames.RetryAfter, "10");

So we add at a given key a string that is implicitly converted to a `StringValues` (may hold multiple values), this by using the `IDictionary.Add()` method that may throw if the key already exists.

By using

    httpContext.Response.Headers.Append(HeaderNames.RetryAfter, "10");

We use an `HeaderDictionaryExtensions` helper that checks if the key already exists and if so retrieves the existing `StringValues` and just add to it the new provided string.

In this file there was no warning because `Headers.Add(header)` uses another interface method, the `ICollection.Add()` method to add a `KeyValuePair<string, StringValues>`, but this method also fails if the key already exists, so I applied the same change.